lib/virtual-fs/src/lib.rs

```diff
@@ -333,7 +336,7 @@ pub trait VirtualFile:
     fn set_len(&mut self, new_size: u64) -> Result<()>;
 
     /// Request deletion of the file
-    fn unlink(&mut self) -> Result<()>;
+    fn unlink<'a>(&'a mut self) -> LocalBoxFuture<'a, Result<()>>;
```

Comment:
The `LocalBoxFuture` here and elsewhere in the trait are problematic.

This means tasks that these futures can never be moved across threads, and hence never just be run in a regular tokio runtime.

Is that really what we want?

I assume this was done to not require shared state, like a clone  of the WasiEnv?

Reply:
We've moved everything over to `BoxFuture` now.